code/Test_definitions/carrier-billing-createPayment.feature

```diff
@@ -0,0 +1,428 @@
+  Scenario: Empty object as request body
+    Given the request body is set to "{}"
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+
+  @create_payment_400.03_other_input_properties_schema_not_compliant
+  # Test other input properties in addition to amountTransaction
+  Scenario Outline: Input property values does not comply with the schema
+    Given the request body property "<input_property>" does not comply with the OAS schema at <oas_spec_schema>
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+    Examples:
+      | input_property          | oas_spec_schema                    |
+      | $.sink                  | /components/schemas/CreatePayment  |
+      | $.sinkCredential        | /components/schemas/SinkCredential |                  |
+
+
+  @create_payment_400.04_required_input_properties_missing
+  Scenario Outline: Required input properties are missing
+    Given the request body property "<input_property>" is not included
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+    Examples:
+      | input_property                                                    |
+      | $.amountTransaction                                               |
+      | $.amountTransaction.paymentAmount                                 |
+      | $.amountTransaction.referenceCode                                 |
+      | $.amountTransaction.paymentAmount.chargingInformation             |
+      | $.amountTransaction.paymentAmount.chargingInformation.amount      |
+      | $.amountTransaction.paymentAmount.chargingInformation.currency    |
+      | $.amountTransaction.paymentAmount.chargingInformation.description |
+
+
+  @create_payment_400.05_clientCorrelator_in_use
+  Scenario: Using the same client correlator for two different payment requests
+    Given the request body property includes property "$.clientCorrelator" with a value already use in a non-completed payment
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+
+  @create_payment_400.06_invalid_sink
+  Scenario: Using a invalid sink value
+    Given the request body property includes property "$.sink" with an HTTP endpoint
```

Comment:
To be sure to get this one: we value here the sink attribute with _something_ which is not an url - correct?

Reply:
It is to guarantee sink is an HTTPs endpoint

As per commonalities:
![image](https://github.com/user-attachments/assets/5ec32bbc-27ff-41cd-b53e-f98a64ae697c)

<html>
<body>
<!--StartFragment-->
sink | string | https callback address where the notification must be POST-ed | mandatory
-- | -- | -- | --


<!--EndFragment-->
</body>
</html>

Maybe better to use inverse wording?

Given the request body property includes property "$.sink" with a non HTTPs endpoint

Reply:
Yes I think it is better to inverse wording:
something like:
`Given the request body property includes property "$.sink" is not set to an url`
is it less confusing no?